### PR TITLE
Add Rector\Rector\Dynamic\ReturnTypehintRector 

### LINKED ISSUE #399: Add Rector\Rector\Dynamic\ReturnTypehintRector
Mentioned in #353 

```yaml
     @todo, use this and change typehint of return value
    Rector\Rector\Dynamic\ReturnTypehintRector:
            $returnTypehintByMethod:
                # OrderInterface::getAdjustmentsRecursively and OrderItemInterface::getAdjustmentsRecursively return types changed from array to Collection
                'Sylius\Component\Order\Model\OrderInterface':
                    'getAdjustmentsRecursively':
                        'array': 'Doctrine\Common\Collections\Collection'

                'Sylius\Component\Order\Model\OrderItemInterface':
                    'getAdjustmentsRecursively':
                        'array': 'Doctrine\Common\Collections\Collection'

                # PrioritizedServiceRegistryInterface::all method return type changed from Zend's PriorityQueue to iterable
                'Sylius\Component\Registry\PrioritizedServiceRegistryInterface':
                    'all':
                        'Zend\Stdlib\PriorityQueue': 'iterable'
```

### PR DESCRIPTION
**Pull request by TomasVotruba**
Closes #399

### CODE REVIEW

**Comment by TomasVotruba** [comment]
Ready to review, the Rector starts here: https://github.com/rectorphp/rector/pull/429/files#diff-16118c59f63336e3b940d16bf88a741c

**Comment by carusogabriel** [comment]
@TomasVotruba Please don't merge before I review :grinning:

**Comment by TomasVotruba** [comment]
I would not dare :wink:

**Comment by TomasVotruba** [comment]
It's ready now

File: src/Naming/PropertyNaming.php [hunk 1]
```diff
@@ -25,6 +25,9 @@ public function underscoreToName(string $underscoreName): string
         return lcfirst($camelCaseName);
     }
 
+    /**
+     * @todo use TypeAnalyzer instead
```
Comment (carusogabriel) [inline src/Naming/PropertyNaming.php]: Can we resolve this one in this PR?

File: src/Php/TypeAnalyzer.php [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+namespace Rector\Php;
+
+use Nette\Utils\Strings;
+
+final class TypeAnalyzer
+{
+    public function isPhpReservedType(string $type): bool
+    {
+        return in_array($type, ['string', 'bool', 'mixed', 'object', 'iterable', 'array'], true);
```
Comment (carusogabriel) [inline src/Php/TypeAnalyzer.php]: Missing types:
- `float`
- `int`

And should we care about `boolean`, `double`? The variants?

File: src/Rector/Dynamic/ReturnTypehintRector.php [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+    }
+
+    public function getDefinition(): RectorDefinition
+    {
+        return new RectorDefinition('[Dynamic] Changes defined return typehint of method and class.', [
+            new CodeSample(
+                <<<'CODE_SAMPLE'
+class SomeClass
+{
+    public getData();
```
Comment (carusogabriel) [inline src/Rector/Dynamic/ReturnTypehintRector.php]: Missing dockblock or it will be set in the `rector.yml`?

Comment (TomasVotruba) [reply src/Rector/Dynamic/ReturnTypehintRector.php]: This rector completes typehints only. Or what do you mean?

Comment (carusogabriel) [reply src/Rector/Dynamic/ReturnTypehintRector.php]: I meant if this Rector will discover the typehint, or will be provided :smile:

File: src/Php/TypeAnalyzer.php (hunk 2 above)
Comment (TomasVotruba) [reply src/Php/TypeAnalyzer.php]: I'll add them, thanks

File: src/Naming/PropertyNaming.php (hunk 1 above)
Comment (TomasVotruba) [reply src/Naming/PropertyNaming.php]: Will try

File: src/Rector/Dynamic/ReturnTypehintRector.php (hunk 3 above)
Comment (TomasVotruba) [reply src/Rector/Dynamic/ReturnTypehintRector.php]: Provided :) It should be clear from tests, isn't it?

Rectors always works with prepared configuration, so they're reliable in this matter. 

Deduction can be pre-process of creating such configs, but that's another story

**Comment by TomasVotruba** [comment]
Resolved

**Pull request merged**